From 552084d4da41833a97d19d12f5ee5c3ba02e400a Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 14 Mar 2020 07:31:34 -0700
Subject: [PATCH V2] selftest/bpf: Use CHECK macro instead of RET_IF

backporting 634efb750435d0a489dc58477d4fcb88b2692942 causes build
failures because RET_IF is defined in 7ee0d4e97b889c0478af9c1a6e5af658b181423f
but that is not backported

Upstream-Status: Submitted
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Cc: Jakub Sitnicki <jakub@cloudflare.com>
Cc: Alexei Starovoitov <ast@kernel.org>
---
v2: Add return on check to get complete logic of RET_IF

 tools/testing/selftests/bpf/test_select_reuseport.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/tools/testing/selftests/bpf/test_select_reuseport.c b/tools/testing/selftests/bpf/test_select_reuseport.c
index cdbbdab2725f..093ef8547abb 100644
--- a/tools/testing/selftests/bpf/test_select_reuseport.c
+++ b/tools/testing/selftests/bpf/test_select_reuseport.c
@@ -616,13 +616,15 @@ static void cleanup_per_test(void)
 
 	for (i = 0; i < NR_RESULTS; i++) {
 		err = bpf_map_update_elem(result_map, &i, &zero, BPF_ANY);
-		RET_IF(err, "reset elem in result_map",
-		       "i:%u err:%d errno:%d\n", i, err, errno);
+		if (CHECK(err, "reset elem in result_map",
+		       "i:%u err:%d errno:%d\n", i, err, errno))
+			return;
 	}
 
 	err = bpf_map_update_elem(linum_map, &zero, &zero, BPF_ANY);
-	RET_IF(err, "reset line number in linum_map", "err:%d errno:%d\n",
-	       err, errno);
+	if (CHECK(err, "reset line number in linum_map", "err:%d errno:%d\n",
+	       err, errno))
+		return;
 
 	for (i = 0; i < REUSEPORT_ARRAY_SIZE; i++)
 		close(sk_fds[i]);
-- 
2.25.1

